features/FEATURE_PSA/supporting_psa_in_mbed-os.md

```diff
@@ -180,6 +181,8 @@ The following example shows a PSA enabled dual-core target, `PSoC64`,
 Please pay attention to the config options `extra_labels_add` and `device_has_remove`. If needed, a PSA target definition **MUST** use [extra_labels/device_has]`_add` or [extra_labels/device_has]`_remove` (not `extra_labels` or `device_has`) to add/remove either extra_labels or target capabilities. Also, use `feature_`[add/remove] to add/remove a feature.
 Check [extra_labels](https://os.mbed.com/docs/mbed-os/v5.14/reference/adding-and-configuring-targets.html), [device_has](https://os.mbed.com/docs/mbed-os/v5.14/reference/adding-and-configuring-targets.html) and [features](https://os.mbed.com/docs/mbed-os/v5.14/reference/adding-and-configuring-targets.html) for more information.
 
+By default TF-M build generates a `bin` file. If the target requires a `hex` file then the attribute `"OUTPUT_EXT": "hex"` should be added to the target definition.
```

Comment:
It's target specific which format should be used for flashing to a device. We shouldn't check in all types just because (hex, bin). We should check in what will be needed. Don't check in stuff you don't need. What you need is target specific. The binaries themselves are also only useful for one target.

Reply:
Other targets, like v8-M ones, will most likely need either a `.a` file, or a `.bin` which we can include from an assembly file, so that we can link the SPE and NSPE images together into one single programmable binary.

Reply:
@jainvikas8 `I think we need to change this statement again, as we don't tend to add .bin any more`. It is not entirely true. The TF-M build process generated a `bin` file by default and we can checkin the same bin file. When generating a complete binary, we can make use of assembler instructions or post-binary hooks to combine the binary files ( bootloader if exists, secure, non-secure). Therefore this will be the default choice.
This approach is not feasible if there is a dependency on third party tools which expects a `hex` file. Adding new attribute `"OUTPUT_EXT": "hex"` is to handle those use cases.

This behaviour is same as that of Mbed OS build tools.